samcli/local/apigw/local_apigw_service.py

```diff
@@ -230,6 +235,17 @@ def _parse_lambda_output(lambda_output, binary_types, flask_request):
 
         return status_code, headers, body
 
+    @staticmethod
+    def _invalid_apig_response(output):
+        allowable = [
```

Comment:
you could make allowable a set to begin with.
`
allowable = {...}
`

Reply:
Changed, though I may have to change further as the set subtraction operator doesn't seem to be supported in Python 2.7

Reply:
I made this change, just had to add the `set()` casting explicitly to the keys.